snapshot-labs/sx-monorepo
# feat: replace favorites by follows (read-only)

### Summary



Closes partially: #282, which is split in multiple PR for easier review

This PR replaces the favorites feature by the the follows system already present on v1. For now, follows are only read-only. This will make onchain spaces not followable until all PR for this feature are merged

- Load an account's follows list from v1 hub
- In the sidebar, show the followed spaces, and keep a local copy to save the sort order
- On all spaces, replace the star/unstar icon by the "follow/unfollow" button. This button is read-only in this PR scope, and will only show different label depending on the space's follow status.
- Make sidebar scrollable

### How to test

1. Log in with an eth wallet
2. Go to any space
5. The "star" button should be replaced by a "Follow/Unfollow" button
6. The button should display different label dependent on whether you're following the space (go to v1 to follow/unfollow the spaces)
7. The sidebar should show your followed spaces
8. Changing the order of the sidebar items should be persisted
9. Log out
12. Log in again with a different ETH wallet
13. Followed spaces for the new wallet should appear in the sidebar
14. Log out and login again with your first wallet
15. It should load followed spaces from the wallet, in the same order as when you logged out

### To-Do

- [x] Change base branch back to master once #217 is merged
- Coming in a later PR will be to add support for write actions: https://github.com/snapshot-labs/sx-monorepo/pull/293
